### PR TITLE
Implemented LuckPerms compat

### PR DESCRIPTION
**Pull request by BrainStone**
This simply adds a new LuckPerms context to be able to distinguish between a normal player and a bedrock player.

### CODE REVIEW

File: plugin/spigot/src/main/java/org/dragonet/plugin/bukkit/compat/luckperms/BedrockClientContextCalculator.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import me.lucko.luckperms.api.context.MutableContextSet;
+import org.bukkit.entity.Player;
+import org.dragonet.plugin.bukkit.DPAddonBukkit;
+
+public class BedrockClientContextCalculator implements ContextCalculator<Player> {
+  @Override
+  public MutableContextSet giveApplicableContext(Player subject,
+      MutableContextSet accumulator) {
+    accumulator.add("bedrock-client", Boolean
+        .toString(DPAddonBukkit.getInstance().bedrockPlayers.contains(subject.getUniqueId())));
```
Comment (HoverEpic) [inline plugin/spigot/src/main/java/org/dragonet/plugin/bukkit/compat/luckperms/BedrockClientContextCalculator.java]: is it working? on bungeecord, the player is registered on login, dur to the special login case. But on Bukkit/Spigot, the player is "registered" after a plugin message, some seconds after login.

Comment (BrainStone) [reply plugin/spigot/src/main/java/org/dragonet/plugin/bukkit/compat/luckperms/BedrockClientContextCalculator.java]: As far as I know this context gets calculated every time it is checked. I'll check back with Luck to confirm though.

Comment (BrainStone) [reply plugin/spigot/src/main/java/org/dragonet/plugin/bukkit/compat/luckperms/BedrockClientContextCalculator.java]: > the result is cached for approx 1 tick
>
> \- Luck

Yes, it is implemented correctly @HoverEpic .

Comment (HoverEpic) [reply plugin/spigot/src/main/java/org/dragonet/plugin/bukkit/compat/luckperms/BedrockClientContextCalculator.java]: Nice

**Pull request merged**